torch/quantization/observer.py

```diff
@@ -137,7 +186,51 @@ def calculate_qparams(self):
 
     @torch.jit.export
     def extra_repr(self):
-        return 'min_val={}, max_val={}'.format(self.min_val, self.max_val)
+        return "min_val={}, max_val={}".format(self.min_val, self.max_val)
+
+
+class PerChannelMinMaxObserver(ObserverBase):
+    r"""Per Channel Observer Module
+    The module will record the running average of max and min value for each
+    channel of the observed Tensor and calculate_qparams will calculate
+    scales and zero_points for each channel
+    """
+
+    __annotations__ = {
+        "min_vals": Optional[torch.Tensor],
+        "max_vals": Optional[torch.Tensor],
+    }
+
+    def __init__(self, ch_axis=0, **kwargs):
+        super(PerChannelMinMaxObserver, self).__init__(**kwargs)
+        self.ch_axis = ch_axis
+        self.min_vals = None
+        self.max_vals = None
+
+    def forward(self, x):
+        min_vals = self.min_vals
```

Comment:
To ensure that autograd does not see this part during training:, please do:
```python
def forward(self, x):
    with torch.no_grad():
 <Body of code> 
```

Reply:
Seems after the change scriptable test would fail with the error:
> torch.jit.frontend.UnsupportedNodeError: with statements aren't supported: